src/SonarScanner.MSBuild.Shim/SonarEngineWrapper.cs

```diff
@@ -102,6 +104,23 @@ private bool Execute(ProcessRunnerArguments args)
                 yield return new ProcessRunnerArguments.Argument(property.AsSonarScannerArg(), true);
             }
         }
+
+        if (!userCmdLineArguments.TryGetValue(SonarProperties.TruststorePassword, out var truststorePassword))
+        {
+            var truststorePath = config.ScannerOptsSettings.FirstOrDefault(x => x.Id == SonarProperties.JavaxNetSslTrustStore);
+            truststorePassword = TruststoreUtils.TruststoreDefaultPassword(truststorePath?.Value, runtime.Logger);
+        }
+
+        if (!SonarPropertiesDefault.TruststorePasswords.Contains(truststorePassword)
+            || scannerOpts is null
+            || !scannerOpts.Contains($"-D{SonarProperties.JavaxNetSslTrustStorePassword}="))
```

Comment:
Do you really mean `||` here? I think this `yield return`s much too often with these `||`'s.

e.g.: Do we ever need to add `-D{SonarProperties.JavaxNetSslTrustStorePassword}` if `truststorePassword` is in `SonarPropertiesDefault.TruststorePasswords`?

Note to myself: UTs/ITs not really reviewed because this feels too noisy.